packages/opds-web-client/src/authMiddleware.ts

```diff
@@ -122,6 +123,8 @@ export default (authPlugins: AuthPlugin[], pathFor: PathFor) => {
                       error,
                       attemptedProvider
                     ));
+                    // Because authProviders is empty for a test.
+                    resolve();
```

Comment:
I don't think we should add this for a test. What should happen if `authProviders` is actually empty?

Reply:
I'm not too sure what this part is testing but it seems like it will just end here without a `resolve` or a `reject`.

Reply:
I still don't understand this. There's a test that breaks if you take this out?

Reply:
These tests https://github.com/NYPL-Simplified/opds-web-client/blob/ac847e1c6072f7da8e6effc2129d99a156ed59a1/packages/opds-web-client/src/__tests__/authMiddleware-test.ts#L114 get hung up because it's waiting to be resolved to go to the `.then()` but if you remove this `resolve`, then the tests just time out. It's not just the test in the link but that test and every single test after it.

Reply:
Okay, I think I get it now. We were leaving an unresolved promise after the authentication had failed - we caught the error from a different promise, but didn't resolve or reject this promise. I think it's okay to resolve since the problem was handled, but I would take out the comment about the test. I think leaving it unresolved also shows errors in the browser console.